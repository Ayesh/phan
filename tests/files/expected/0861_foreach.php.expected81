%s:4 PhanParamSignaturePHPDocMismatchReturnType Declaration of real/@method function current() should be compatible with real/@method function current() : object (method returning 'T' cannot override method returning 'object') defined in internal:0
%s:16 PhanTypeMismatchReturn Returning $elem of type \ArrayObject but test() is declared to return string
