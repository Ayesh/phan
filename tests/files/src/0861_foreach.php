<?php
/**
 * @template T
 * @method T current() TODO: Support parent types of templates
 * @suppress PhanParamSignaturePHPDocMismatchReturnType TODO: Support parent types of templates
 */
class Set extends \SplObjectStorage
{
}
/**
 * @param Set<ArrayObject> $arrays
 * @return string
 */
function test(Set $arrays) {
    foreach ($arrays as $elem) {
        return $elem;
    }
    return 'default';
}
