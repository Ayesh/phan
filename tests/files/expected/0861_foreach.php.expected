%s:16 PhanTypeMismatchReturn Returning $elem of type \ArrayObject but test() is declared to return string
